src/main/features/claimant-response/routes/payment-option.ts

```diff
@@ -2,25 +2,147 @@ import * as express from 'express'
 
 import { AbstractPaymentOptionPage } from 'shared/components/payment-intention/payment-option'
 import { AbstractModelAccessor, DefaultModelAccessor } from 'shared/components/model-accessor'
-import { PaymentIntention } from 'shared/components/payment-intention/model/paymentIntention'
+import { PaymentIntention as DraftPaymentIntention } from 'shared/components/payment-intention/model/paymentIntention'
+import { PaymentIntention } from 'claims/models/response/core/paymentIntention'
 
 import { DraftClaimantResponse } from 'claimant-response/draft/draftClaimantResponse'
 
 import { claimantResponsePath, Paths } from 'claimant-response/paths'
+import { Claim } from 'claims/models/claim'
+import { FullAdmissionResponse } from 'claims/models/response/fullAdmissionResponse'
+import { PartialAdmissionResponse } from 'claims/models/response/partialAdmissionResponse'
+import { CourtDecisionHelper } from 'shared/helpers/CourtDecisionHelper'
+import { DecisionType } from 'common/court-calculations/courtDecision'
+import { PaymentOption } from 'claims/models/paymentOption'
+import { PaymentPlan } from 'common/payment-plan/paymentPlan'
+import { PaymentPlanHelper } from 'shared/helpers/paymentPlanHelper'
+import { Frequency } from 'common/frequency/frequency'
+import { User } from 'idam/user'
+import { MomentFactory } from 'shared/momentFactory'
+import { PaymentType } from 'shared/components/payment-intention/model/paymentOption'
+import { Draft } from '@hmcts/draft-store-client'
 
-class PaymentOptionPage extends AbstractPaymentOptionPage<DraftClaimantResponse> {
+export class PaymentOptionPage extends AbstractPaymentOptionPage<DraftClaimantResponse> {
+
+  static generateCourtOfferedPaymentIntention (draft: DraftClaimantResponse, claim: Claim, decisionType: DecisionType): PaymentIntention {
+    const courtOfferedPaymentIntention = new PaymentIntention()
+    const claimResponse = claim.response as FullAdmissionResponse | PartialAdmissionResponse
+
+    if (decisionType === DecisionType.CLAIMANT || decisionType === DecisionType.CLAIMANT_IN_FAVOUR_OF_DEFENDANT) {
+      if (draft.alternatePaymentMethod.paymentOption.option.value === PaymentOption.IMMEDIATELY) {
+        courtOfferedPaymentIntention.paymentOption = PaymentOption.IMMEDIATELY
+        courtOfferedPaymentIntention.paymentDate = MomentFactory.currentDate().add(5,'days')
+      }
+      return courtOfferedPaymentIntention
```

Comment:
Should we be returning an empty intention here, or `undefined`?

Reply:
And for the other two decision type blocks, can we explicitly return `undefined` for the "immediately" payment option? I think the rest of the code guarantees it won't happen, but this method doesn't know that and it might look cleaner.